var/spack/repos/builtin/packages/tandem/package.py

```diff
@@ -61,14 +62,41 @@ class Tandem(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("eigen@3.4.0")
 
     depends_on("zlib-api")
-    depends_on("petsc@3.14.6:3.18.5 +int64 +mumps +scalapack memalign=32")
-    depends_on("petsc@3.14.6:3.18.5 +int64 +mumps +scalapack +knl", when="target=skylake:")
-    depends_on("petsc@3.14.6:3.18.5 +int64 +mumps +scalapack memalign=32 +cuda", when="+cuda")
-    depends_on("petsc@3.14.6:3.18.5 +int64 +mumps +scalapack memalign=32 +rocm", when="+rocm")
+    depends_on("petsc@3.14.6: +int64 +mumps +scalapack memalign=32")
+    depends_on("petsc@3.14.6: +int64 +mumps +scalapack +knl", when="target=skylake:")
```

Comment:
Others can be simplified:
```suggestion
    depends_on("petsc +knl", when="target=skylake:")
```